src/StaticFAQ/FAQArticleDetail.js

```diff
@@ -44,21 +65,45 @@ class FAQArticleDetail extends React.Component<Props> {
       return <div>{params.error.message}</div>;
     }
     if (params.props) {
-      return <FAQArticleDetailContent article={params.props.FAQArticle} />;
+      return (
+        <React.Fragment>
+          <CustomBreadcrumbs
+            breadcrumbs={
+              params.props.FAQCategory
+                ? [{ title: 'Home' }]
+                    .concat(params.props.FAQCategory.ancestors)
+                    .concat(params.props.FAQCategory)
+                    .concat([{ title: params.props.FAQArticle.title }])
+                : [{ title: 'Search' }].concat([
+                    { title: params.props.FAQArticle.title },
+                  ])
+            }
+          />
+          <FAQArticleDetailContent article={params.props.FAQArticle} />
+        </React.Fragment>
+      );
     }
 
     return <Loader />;
   };
 
   render() {
     const articleId = idx(this.props.match, _ => _.params.articleId);
+    const categoryId = idx(this.props.match, _ => _.params.categoryId);
 
     return (
       <div className="faq-article-detail">
         {!this.props.user && <ContentHeader />}
         <QueryRenderer
-          query={queryFAQArticleDetail}
-          variables={{ id: articleId }}
+          query={
+            categoryId === 'search'
+              ? queryFAQArticleDetailSearchResult
+              : queryFAQArticleDetail
+          }
+          variables={{
+            id: articleId,
+            ...(categoryId === 'search' ? {} : { category_id: categoryId }),
```

Comment:
This is repeating the same conditional as in L99. Can you just check `if(categoriId==='search')` *once* and then throw one `QueryRenderer` or another depending on the case. Just like in [here](https://github.com/kiwicom/smart-faq/blob/master/src/StaticFAQ/FAQCategoryList.js#L157)